lib/ansible/module_utils/aws/core.py

```diff
@@ -158,3 +160,18 @@ def fail_json_aws(self, exception, msg=None):
         else:
             self._module.fail_json(msg=message, exception=last_traceback,
                                    **camel_dict_to_snake_dict(response))
+
+
+class _RetryingBotoClientWrapper:
```

Comment:
It's pretty magical (in both good and bad senses!). 

Your worry is reasonable, I've accidentally hit infinite loops even when knowing I'm hitting AWSRetry, this might definitely cause some problems for module development down the line. I do like the idea of not having to write a separate method just for backoffs though!

Reply:
What about if we make it more explicit at the call-site without still requiring a wrapping method. E.g. something like:

    # Setup Connection
    connection = module.client('ses', retry_decorator=AWSRetry.jittered_backoff)

    # Usage something like
    connection.list_identity_policies(Identity=identity, aws_retry=True)

    # or 
    connection.list_identity_policies(Identity=identity, aws_retry={retries:5, delay:5})

Or possibly even just passing the whole retry function as a parameter to the call e.g.

    # Setup Connection and opt into the decorator
    connection = module.client('ses', retry_decorator_enabled=True)

    # Usage something like
    connection.list_identity_policies(Identity=identity, retry_decorator=AWSRetry.jittered_backoff)

Pretty sure I could make either of those syntaxes work, and they'd make it explicit while still keeping the explicitness of the retry.

Reply:
@ryansb unless I'm miss-understanding you, it's already taking the parameters I just happen to be using the defaults in this particular case. Function call is made [here](https://github.com/ansible/ansible/pull/36623/files#diff-ed4981862e93fb5f5c08b676cfe95aa8R183) and can take parameters.

I like the idea of adding a boolean "do-retry" at each call-site to keep it explicit, I'll add that in.

Reply:
Yup, you got it. :1st_place_medal:

Reply:
Implementation has been updated to @ryansb 's suggestion.

i.e.
* accept the generated decorator when obtaining a connection so we can specify the retry count and other parameters in one place
* at each call-site specify `aws_retry=True` to opt in to the retrying behaviour. If this arg is false or omitted use the unwrapped boto function so there's no retry